### DISCUSSION TITLE
Only post-processing?

### DISCUSSION THREAD

**Discussion by RoosvOeveren** [muscbridge/PyDKE#313] [Technical Support]
Hi PyDesigner community,

Is it possible to run PyDesigner without any pre-processing at all? I have multi-shell data that has already been preprocessed, from which I would like to obtain DTI and DKI metrics.

I ran PyDesigner on my raw data ('pydesigner -s --verbose -o [output directory] [xxx.nii]'), which resulted in all the expected outcome measures. However, brain distortions in these outcome measures (e.g. in dti_fa maps) are worse than in my own pre-processed data.

I tried simply running PyDesigner with my pre-processed data, without any pre-processing flag ('pydesigner --verbose -o [output directory] [xxx.nii]'). The outputs however are noisy DTI metrics and no DKI at all. 

Is there a command line or flag that enables only post-processing?

Thank you in advance,
Roos

**Comment by gonzoBlackMamba** [comment]
Hey @RoosvOeveren,

Sorry to hear about the issue. Let me see if I understand. You have data already pre-processed and want to do the tensor fitting etc., right? In the main folder that you download from github, there should be a .ipyn notebook called Tensor Fitting Guide that gives some direction on how one could do that. If you could try that and let us know. I do know that there is currently no way to run pyD and start it after the pre-processing at this time, but it would be a nice feature to include so we will add it to our to-do list. Thanks for using and testing pyD out!

-Hunter

**Comment by TheJaeger** [comment]
@RoosvOeveren running PyDesigner without any pre-processing flags i.e. `pydesigner --verbose -o $OUTPUT_DIR $DWI.nii` will bypass all pre-processing steps and advance to just extracting diffusion metrics. That's the behavior on the [latest release v1.0.0](https://github.com/muscbridge/PyDesigner/releases/tag/v1.0.0).

Parameter maps are highly dependent on the quality of pre-processing. In particular, Eddy current correction alongside co-registration have the greatest impact on the quality of parameter maps. I would highly recommend PyDesigner's pre-processing pipeline as it goes through the appropriate steps to maximize SNR and thereby yield cleaner parameter maps.

Regarding the absence of DKI metrics, would you mind posting your BVAL file? It could be that b-values are close to PyDesigner's DKI threshold but not quite there. In which case, we can create a hotfix to improve detection of DKI datasets.
